Suggestions from Rohan's review (#6)
dconnolly/draft-connolly-cfrg-hpke-mlkem: draft-connolly-cfrg-hpke-mlkem.md

--- abstract

> This document defines Module-Lattice-Based Key-Encapsulation Mechanism
> (ML-KEM) options for Hybrid Public-Key Encryption (HPKE). ML-KEM is

Reviewer: ```suggestion
This document defines Module-Lattice-Based Key-Encapsulation Mechanism
(ML-KEM) KEM options for Hybrid Public-Key Encryption (HPKE). ML-KEM is
```

> This document defines Module-Lattice-Based Key-Encapsulation Mechanism
> (ML-KEM) KEM options for Hybrid Public-Key Encryption (HPKE). ML-KEM is

Reviewer: ```suggestion
This document defines Module-Lattice-Based Key-Encapsulation Mechanism
(ML-KEM) as defined in {{FIPS203}} KEM options for Hybrid Public-Key Encryption (HPKE). ML-KEM is
```

> This document defines Module-Lattice-Based Key-Encapsulation Mechanism
> (ML-KEM) as defined in {{FIPS203}} KEM options for Hybrid Public-Key Encryption (HPKE). ML-KEM is

Reviewer: ```suggestion
This document defines Module-Lattice-Based Key-Encapsulation Mechanism
(ML-KEM) KEM options for Hybrid Public-Key Encryption (HPKE). ML-KEM is
```

## Motivation {#motivation}

ML-KEM {{FIPS203}} is a Key-Encapsulation Mechanism (KEM) which is believed
to be secure against both classical and quantum cryptographic attacks. For
parties that must move to exclusively post-quantum algorithms, this document
defines pure post-quantum algorithms for the Hybrid Public-Key Encryption
(HPKE) protocol {{RFC9180}}. ML-KEM as a post-quantum IND-CCA2-secure KEM
fits nicely into HPKE's design. Supporting multiple security levels for
ML-KEM allows a spectrum of use cases including settings where the (United
States) National Institute of Standards (NIST) security category 5 is
required.

## Not an authenticated KEM {#S-notauth}

ML-KEM is a plain KEM that does not support the static-static key exchange
that allows HPKE based on Diffie-Hellman (DH) based KEMs and their (optional)
authenticated modes.

# Conventions and Definitions {#conventions}

{::boilerplate bcp14-tagged}

`GenerateKeyPair`, `DeriveKeyPair`, `SerializePublicKey`,
`DeserializePublicKey`, `Encap`, `Decap`, `AuthEncap`, `AuthDecap`,
`Nsecret`, `Nenc`, `Npk`, and `Nsk` are defined in Section 4 of {{RFC9180}}.

> When used in the Security Consideration section, `PK` refers to public key
> and `CT` refers to cipher text.

Reviewer: ```suggestion
When used in the Security Consideration section, `PK` refers to public key
and `CT` refers to ciphertext as modeled in {{CDM23}}.
```

**TODO**: explain or reference IND-CCA, IND-CCA2, MAL-BIND-K-PK,
MAL-BIND-K-CT, and LEAK-BIND-K-PK.

# Usage {#usage}

{{FIPS203}} supports two different key formats. This document only
supports the 64-byte seed `(d, z)`. This format supports stronger binding
properties for ML-KEM than the expanded format. The 64-byte seed format
protects against re-encapsulation attacks. This format provides properties
closer to the generic DHKEM binding properties defined in Section 4.1 of
{{RFC9180}}.

The encapsulation and decapsulation keys are computed, serialized, and
deserialized as described in {{FIPS203}} where the decapsulation keys MUST
be in the 64-byte `(d, z)` format. The 'expanded' format where the
decapsulation key is expanded into a variable size based on the parameter
set but includes the hash of the encapsulation key is not used.

> Unlike DHKEM, ML-KEM does not rely on an external Key-Derivation Function
> (KDF).

Reviewer: Removing, most KEMs are more like ML-KEM than like DHKEM

```suggestion
```

properties of the ciphersuite KEM's X-BIND-K-CT and X-BIND-K-PK
properties. These computational binding properties for KEMs were formalized
in {{CDM23}}. {{CDM23}} describes DHKEM as MAL-BIND-K-PK and MAL-BIND-K-CT

> secure as a result of the inclusion of the serialized DH public keys (the KEM's
> PK and CT) in the DHKEM Key Derivation Function (KDF). MAL-BIND-K-PK and
> MAL-BIND-K-CT security ensures that the shared secret K 'binds' (is uniquely determined by) the encapsulation key and/or the ciphertext, even when the
> adversary is able to create or modify the key pairs or has access to
> honestly-generated leaked key material.

Reviewer: ```suggestion
secure as a result of the inclusion of the serialized DH public keys (the KEM's
PK and CT) in the DHKEM Key Derivation Function (KDF). MAL-BIND-K-PK and
MAL-BIND-K-CT security ensures that the shared secret K 'binds' (is
uniquely determined by) the encapsulation key and/or the ciphertext,
even when the adversary is able to create or modify the key pairs or
has access to honestly-generated leaked key material.
```

properties of the ciphersuite KEM's X-BIND-K-CT and X-BIND-K-PK
properties. These computational binding properties for KEMs were formalized
in {{CDM23}}. {{CDM23}} describes DHKEM as MAL-BIND-K-PK and MAL-BIND-K-CT

> secure as a result of the inclusion of the serialized DH public keys (the KEM's
> PK and CT) in the DHKEM Key Derivation Function (KDF). MAL-BIND-K-PK and
> MAL-BIND-K-CT security ensures that the shared secret 'binds' or uniquely
> determines the encapsulation key and the encapsulated shared secret
> ciphertext, even when the adversary is able to create or modify the key
> pairs or has access to honestly-generated leaked key material.

Reviewer: ```suggestion
secure as a result of the inclusion of the serialized DH public keys (the KEM's
PK and CT) in the DHKEM Key Derivation Function (KDF). MAL-BIND-K-PK and
MAL-BIND-K-CT security ensures that the shared secret K 'binds' (is uniquely determined by) the encapsulation key and/or the ciphertext, even when the
adversary is able to create or modify the key pairs or has access to
honestly-generated leaked key material.
```
